Improve CI

w-masahiro-ct opened the pull request:
I improved CI.

gjtorikian commented:
Ah, unfortunately, I can't accept the changes to `setup-ruby`. I like to ensure all my projects use the same actions I created so I don't have to think about it. Besides which, `setup-languages` already performs these operations: https://github.com/yettoapp/actions/blob/abbd75422c1118794948f6cc682efe5a7cf988b0/setup-languages/action.yml#L33-L40

Also, why delete the `.ruby-version`?

w-masahiro-ct commented:
>I like to ensure all my projects use the same actions I created so I don't have to think about it.

I changed to use yettoapp/actions/setup-languages.

>Also, why delete the .ruby-version?

Repositories of commonly used Gems such as Puma, Faraday, and RuboCop do not contain .ruby-version file.
I think Ruby version should not be specified in the Gem repository, so that is why I added matrix to CI.
However, it's not a big problem, so it's okay to restore the .ruby-version file👌

w-masahiro-ct commented:
For reference, below is the .gitignore for Ruby from github/gitignore. https://github.com/github/gitignore/blob/6eeebe6f49678aacd8311ce079842c971b3ebe96/Ruby.gitignore#L46-L50

gjtorikian commented:
Oh, that's really interesting. In all my years of Ruby I never knew that was suggested.

But, what if the code in the project is written using some new Ruby syntax, that the contributor doesn't have installed? In other words, how would I communicate to someone else that "Ruby 3.x.y is recommended to work with this project"?

w-masahiro-ct commented:
How about using TargetRubyVersion in .rubocop.yml?
I just commited now.
Ruby version of this repository is 3.1, so I specified 3.1 in .rubocop.yml.
For reference, below is the .rubocop.yml of Faraday.
https://github.com/lostisland/faraday/blob/a9cf00425e3abc99b78952af44deb2912a65a882/.rubocop.yml#L10

gjtorikian commented:
Yeah...Faraday has over 5,000 stars and hundreds of millions of downloads. I consider this (and most of my projects) to be repositories that meet my needs, and my tooling. If the contribution and usage pool increases to the point where such formal versioning statements are needed, that's great. 

But for now, I like having all my projects formatted the same way (even if they're "wrong") so I have one less thing to think about. ✌️ In any event, it doesn't look like the lack of such versioning prevented earlier contributions, so I do not think this should be done. Thank you for your help and feedback though!

w-masahiro-ct commented:
@gjtorikian Sorry for reopening PR #45.
I only added the matrix to test CI.
If you also oppose adding the matrix, feel free to close PR #45.
However, In my opinion, it is very important to have a matrix.
